source/extensions/filters/http/jwt_authn/matcher.cc

```diff
@@ -188,6 +189,41 @@ class PathSeparatedPrefixMatcherImpl : public BaseMatcherImpl {
 };
 } // namespace
 
+class PathMatchPolicyMatcherImpl : public BaseMatcherImpl {
+public:
+  PathMatchPolicyMatcherImpl(const RequirementRule& rule,
+                             Server::Configuration::CommonFactoryContext& context)
+      : BaseMatcherImpl(rule, context) {
+    auto& factory = Config::Utility::getAndCheckFactory<Router::PathMatcherFactory>(
+        rule.match().path_match_policy());
+    ProtobufTypes::MessagePtr config = Envoy::Config::Utility::translateAnyToFactoryConfig(
+        rule.match().path_match_policy().typed_config(),
+        ProtobufMessage::getStrictValidationVisitor(), factory);
+
+    absl::StatusOr<Router::PathMatcherSharedPtr> matcher = factory.createPathMatcher(*config);
+
+    if (!matcher.ok()) {
+      throw EnvoyException(std::string(matcher.status().message()));
```

Comment:
This `EnvoyException` is not tested currently, right? Will you be able to have a quick test cover it?

Reply:
Hey, I added a test with wrong configuration to test this status check.
I'm not sure what exactly you mean by also testing the prevention mechanism for nullptr access - it is tested right now, right? Early failure prevents reaching the `match` call on `uri_template_matcher_`.

Reply:
yea, early failure is what i initially suggested in this comment https://github.com/envoyproxy/envoy/pull/35072#discussion_r1681196576 and what I hope to have test coverage here.

LG, Thanks for adding the test!